pvlib/scaling.py

```diff
@@ -0,0 +1,241 @@
+    packages built for cartographic projection such as pyproj (e.g.
+    pyproj.transform()) [2].
+
+    Parameters
+    ----------
+
+    coordinates : numeric
+        Array or list of (latitude, longitude) coordinate pairs. Use decimal
+        degrees notation.
+
+    Returns
+    -------
+    xypos : numeric
+        Array of coordinate distances as (x,y) pairs representing the
+        easting, northing of the position in meters [m].
+
+    References
+    ----------
+    [1] H. Moritz. Geodetic Reference System 1980, Journal of Geodesy, vol. 74,
+    no. 1, pp 128–133, 2000.
+
+    [2] https://pypi.org/project/pyproj/
+
+    [3] Wavelet Variability Model - Matlab Code:
+    https://pvpmc.sandia.gov/applications/wavelet-variability-model/
+    """
+
+    # Added by Joe Ranalli (@jranalli), Penn State Hazleton, 2019
+
+    r_earth = 6371008.7714  # mean radius of Earth, in meters
+    m_per_deg_lat = r_earth * np.pi / 180
+    try:
+        meanlat = np.mean([lat for (lat, lon) in coordinates])  # Mean latitude
+    except TypeError:  # Assume it's a single value?
+        meanlat = coordinates[0]
+    m_per_deg_lon = r_earth * np.cos(np.pi/180 * meanlat) * np.pi/180
+
+    # Conversion
+    pos = coordinates * np.array(m_per_deg_lat, m_per_deg_lon)
+
+    # reshape as (x,y) pairs to return
+    try:
+        return np.column_stack([pos[:, 1], pos[:, 0]])
+    except IndexError:  # Assume it's a single value, which has a 1D shape
+        return np.array((pos[1], pos[0]))
+
+
+def _compute_wavelet(clearsky_index, dt=None):
+    """
+    Compute the wavelet transform on the input clear_sky time series.
+
+    Parameters
+    ----------
+    clearsky_index : numeric or pandas.Series
+        Clear Sky Index time series that will be smoothed.
+
+    dt : float, default None
+        The time series time delta. By default, is inferred from the
+        clearsky_index. Must be specified for a time series that doesn't
+        include an index. Units of seconds [s].
+
+    Returns
+    -------
+    wavelet: numeric
+        The individual wavelets for the time series
+
+    tmscales: numeric
+        The timescales associated with the wavelets in seconds [s]
+
+    References
+    ----------
+    [1] M. Lave, J. Kleissl and J.S. Stein. A Wavelet-Based Variability
+    Model (WVM) for Solar PV Power Plants. IEEE Transactions on Sustainable
+    Energy, vol. 4, no. 2, pp. 501-509, 2013.
+
+    [3] Wavelet Variability Model - Matlab Code:
+    https://pvpmc.sandia.gov/applications/wavelet-variability-model/
+    """
+
+    # Added by Joe Ranalli (@jranalli), Penn State Hazleton, 2019
+
+    try:  # Assume it's a pandas type
+        vals = clearsky_index.values.flatten()
+        try:  # Assume it's a time series type index
+            dt = (clearsky_index.index[1] - clearsky_index.index[0]).seconds
+        except AttributeError:  # It must just be a numeric index
+            dt = (clearsky_index.index[1] - clearsky_index.index[0])
```

Comment:
It's good practice is to put the minimum amount of code in each block of a `try/except/else/finally`, so I suggest moving the pandas dt lookup to an `else` block as shown below

```python
    try:  # Assume it's a pandas type
        vals = clearsky_index.values.flatten()
    except AttributeError:  # Assume it's a numpy type
        vals = clearsky_index.flatten()
        if dt is None:
            raise ValueError("dt must be specified for numpy type inputs.")
    else:  # it is a pandas type, now get the dt
        try:  # Assume it's a time series type index
            dt = (clearsky_index.index[1] - clearsky_index.index[0]).seconds
        except AttributeError:  # It must just be a numeric index
            dt = (clearsky_index.index[1] - clearsky_index.index[0])
```